src/lib/repository/DeviceRepository.js

```diff
@@ -175,6 +179,31 @@ class DeviceRepository {
     return result.result;
   };
 
+  getVariableValue = async (
+    deviceID: string,
+    userID: string,
+    varName: string,
+  ): Promise<Object> => {
+    if (!this._deviceAttributeRepository.doesUserHaveAccess(deviceID, userID)) {
```

Comment:
don't like that we have to do it in every deviceRepo's method and pass userID to every method. maybe do something like `deviceRepo.checkAccess(userID)` and use it inside controller before other calls?